### PR TITLE
feat: JHA-ContactInfo country/region select

### PR DESCRIPTION
**Pull request by Ian-c-cameron**
### Please check if the PR fulfills these requirements:

- [X] After these changes, the app was run and still works as expected
- [ ] Unit tests for these changes were added (if applicable)
- [X] All existing unit tests were run and still pass

### Please specify the type of change this PR introduces (Bug fix, feature addition, content update, chore, etc.):
Changed country and region(if in canada) to CountrySelect and RegionSelect components.

### CODE REVIEW

**Review by harrymaynard-maximus — COMMENTED** [review]
You'll also have to reset the province field when the country changes. You can use a watcher for this.

File: jha/src/views/enrolment/ContactInfoPage.vue [hunk 1]
```diff
@@ -424,6 +435,21 @@ export default {
...
+      if (this.mailCountryOld !== "_") {
+        if(this.mailCountryOld === "Canada" && value !== "Canada") {
+          // don't keep a Canadian province if not in canada
+          this.mailProvince = "";
+        } else if (this.mailCountryOld !== "Canada" && value === "Canada") {
+          // if changing to canada, a canadian province must be selected by default.
+          this.mailProvince = "British Columbia";
+        }
+      }
+      this.mailCountryOld = value;
```
Comment (harrymaynard-maximus) [inline jha/src/views/enrolment/ContactInfoPage.vue:450]: You can get the old value via a second parameter to the watch function. https://vuejs.org/v2/api/#watch

**Review by harrymaynard-maximus — CHANGES_REQUESTED** [review]

File: jha/src/views/enrolment/ContactInfoPage.vue (hunk 1 above)
Comment (Ian-c-cameron) [reply jha/src/views/enrolment/ContactInfoPage.vue:450]: Updated with second parameter

File: jha/src/views/enrolment/ContactInfoPage.vue [hunk 2]
```diff
@@ -424,6 +434,20 @@ export default {
       this.isMailSame = false;
     },
   },
+  watch: {
+    mailCountry: function(value, oldValue) {
+      // don't clear province if the change is from loading from store
+      if (oldValue !== "_") {
```
Comment (harrymaynard-maximus) [inline jha/src/views/enrolment/ContactInfoPage.vue:440]: Just saw this. I don't think `mailCountry` will ever be `"_"`, as it gets overridden by the store value ("Canada"). You can probably remove this check. I think the initial value for `mailCountry` should be `null` anyway.

**Review by harrymaynard-maximus — CHANGES_REQUESTED** [review]

File: jha/src/views/enrolment/ContactInfoPage.vue (hunk 2 above)
Comment (Ian-c-cameron) [reply jha/src/views/enrolment/ContactInfoPage.vue:440]: No it is needed, if they navigate to a previous page and then return, then the country and province will be loaded from the store, if this check isn't there, then the watcher will trigger when it loads from store and overwrite the province field with either "British Columbia"(if stored country is Canada) or ""(if stored country is anything else).  Watchers are triggered when the value is changed from the default set in data during the created function.

Comment (harrymaynard-maximus) [reply jha/src/views/enrolment/ContactInfoPage.vue:440]: Ok, for that case. We usually use a `isPageLoaded` flag, and only make the assignments in the watch function when the flag is `true`.

Here's an example on the Personal Info page: https://github.com/bcgov/MOH-MSP-Enrolment/blob/main/jha/src/views/enrolment/PersonalInfoPage.vue#L860

**Review by harrymaynard-maximus — APPROVED** [review]
Looks good.

**Pull request merged**